packages/workflow-compile/legacy/index.js

```diff
@@ -3,15 +3,37 @@ const fse = require("fs-extra");
 const externalCompile = require("@truffle/external-compile");
 const solcCompile = require("@truffle/compile-solidity");
 const vyperCompile = require("@truffle/compile-vyper");
-const { prepareConfig } = require("../utils");
 const { Shims } = require("@truffle/compile-common");
+const expect = require("@truffle/expect");
+const Config = require("@truffle/config");
+const Artifactor = require("@truffle/artifactor");
+const Resolver = require("@truffle/resolver");
 
 const SUPPORTED_COMPILERS = {
   solc: solcCompile,
   vyper: vyperCompile,
   external: externalCompile
 };
 
+function prepareConfig(options) {
+  expect.options(options, ["contracts_build_directory"]);
+
+  expect.one(options, ["contracts_directory", "files"]);
+
+  // Use a config object to ensure we get the default sources.
+  const config = Config.default().merge(options);
+
+  config.compilersInfo = {};
+
+  if (!config.resolver) config.resolver = new Resolver(config);
+
+  if (!config.artifactor) {
+    config.artifactor = new Artifactor(config.contracts_build_directory);
+  }
+
+  return config;
+}
+
```

Comment:
This appears to be identical to `prepareConfig` in utils.js. Maybe this legacy file can also be ts so it can import instead? Save some imports and duplicate code.

Reply:
So the legacy stuff is eventually going to be removed. I think it might be a breaking change to do so currently because some people might be using it. My approach was to leave it alone quarantined in its own folder (as it is sort of deprecated) as there doesn't seem to be much value in updating it if it is going to be removed the next time we are able to create a breaking change. What do you think?

Reply:
That's fair. I guess we just have to make a note of this and chuck it when workflow-compile is ready for a major bump.